Migrate to catcher_2

ThexXTURBOXx opened the pull request:
# This has now been released as part of [`catcher_2`](https://pub.dev/packages/catcher_2)

You can use it like this:

```yml
dependencies:
  catcher_2: ^1.0.0-alpha.0
```

**For future changes, feel free to migrate to my continuation of this project, now known as `catcher_2`!**

## Related stuff

This PR fixes many current issues in the implementation and updates the package and example dependencies.

Fixes #209
Fixes #213
Fixes #230
Fixes #232
Fixes #238 
Closes #211
Closes #239
Closes #241
Closes #242
Closes #244
Closes #245
Fixes #247 
Fixes #248 
Fixes #249 
Fixes #254
Closes #255
Fixes #257
Fixes #258
Closes #259

solidmetanoia committed:
allow setting explicit SMTP username when it is not the same as email (eg.: AWS SES)

solidmetanoia committed:
updating with EmailAutoHandler parameter info

ThexXTURBOXx committed:
Update Gradle wrapper

ThexXTURBOXx committed:
Update example dependencies

ThexXTURBOXx committed:
Update device_info_plus

ThexXTURBOXx committed:
Bump version

l-giza commented:
@ThexXTURBOXx could you also bump "fluttertoast" reference to 8.1.1? This would remediate its issues with gradle >= 1.5.x

ThexXTURBOXx commented:
@l-giza fluttertoast should already allow `8.1.1` because of its `^` modifier

l-giza commented:
Ah, makes sense. Thanks!

ThexXTURBOXx commented:
catcher `0.7.0` should already support fluttertoast `8.1.1`

narumi147 commented:
can I merge it into my folk?

ThexXTURBOXx commented:
@narumi147 Of course :)

narumi147 commented:
Thanks! merged.

ThexXTURBOXx committed:
Update dependencies

ThexXTURBOXx committed:
Also update dio

ThexXTURBOXx committed:
Update dio properly

ThexXTURBOXx committed:
Update sentry

ThexXTURBOXx committed:
Fix Stack Trace sent to Sentry

ThexXTURBOXx committed:
Fix Flutter 3.10-pre implementation

ThexXTURBOXx commented:
I have updated the description and title of this PR in order to reflect the new changes

ThexXTURBOXx committed:
Better README

ThexXTURBOXx committed:
Update Changelog

ThexXTURBOXx committed:
Migrate namespace declaration

calcitem commented:
Dear @jhomlala,

First of all, thank you for creating this incredibly useful library, Catcher. Recently, I noticed that the project has not been maintained for quite some time. I found that @ThexXTURBOXx has submitted a pull request (#234) addressing several issues and updating dependencies, making Catcher compatible with the latest Flutter version.

Considering the value this PR brings to the project, I kindly request you to review and merge it if possible. If you're unable to maintain the project yourself due to time constraints or other reasons, please consider adding active contributors like @ThexXTURBOXx to the list of collaborators or allowing them to start a Catcher2 project.

This would ensure that the project stays up-to-date and continues to benefit the community. Your contributions have been greatly appreciated, and we hope to see the project thrive with your help or the help of new collaborators.

Thank you for your understanding and support.

Best regards,
Calcitem

ThexXTURBOXx committed:
Update dependencies

ThexXTURBOXx committed:
Fix SDK 3.x support

ThexXTURBOXx committed:
Update constraints

ThexXTURBOXx committed:
Fix implementation

mrclauss commented:
Hey @ThexXTURBOXx , I tried using your fork, but run into a build issue on android:

D:\workspace\contactual\flutter-app\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java:24: error: cannot find symbol
      flutterEngine.getPlugins().add(new com.jhomlala.catcher.CatcherPlugin());
                                                             ^
  symbol:   class CatcherPlugin
  location: package com.jhomlala.catcher
D:\workspace\contactual\flutter-app\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java:69: error: cannot find symbol
      flutterEngine.getPlugins().add(new io.github.ponnamkarthik.toast.fluttertoast.FlutterToastPlugin());
                                                                                   ^
  symbol:   class FlutterToastPlugin
  location: package io.github.ponnamkarthik.toast.fluttertoast
2 errors

FAILURE: Build failed with an exception.

Any clue what's wrong there? Before I start digging deeper into this...

ThexXTURBOXx commented:
That's weird. Did you maybe try a `flutter clean`? Sometimes, that helps wonders :)

ThexXTURBOXx commented:
No problem, nice to hear! :)

fredgrott commented:
I would be nice to have this very useful PR merged since I am writing a full catcher usage guide based on this PR.

MOPineyro commented:
Any chance this will be merged in soon? See the example project is what is failing CI.

ThexXTURBOXx commented:
@MOPineyro The example project is not failing the CI. The CI was just outdated at the time of it running through, so it could not install the newest version of Flutter properly

DanMossa commented:
@ThexXTURBOXx  Could you update flutter_mailer to the latest version to support AGP 8? Thanks!

ThexXTURBOXx committed:
Update dependencies and Flutter again

ThexXTURBOXx committed:
Migrate to new CirrusLabs containers

ThexXTURBOXx committed:
Fix removed command

ThexXTURBOXx commented:
CI has been fixed by migrating to the new CirrusLabs container

haynar commented:
Is there any chance this will be merged any time soon?

ThexXTURBOXx commented:
@haynar Just use the dependency override as described for the time being

Guo2299 commented:
I used the import https://github.com/ThexXTURBOXx/catcher.git. It will report that the sentry library uses 7.2.0 and is incompatible with uuid: ^4.0.0. Can you help me?

ThexXTURBOXx commented:
I am currently at a conference. I will see tomorrow if there is something I can do about it. Most likely, some dependency has a dependency conflict with another library that you use.

ThexXTURBOXx commented:
@Guo2299 It is due to sentry: https://github.com/getsentry/sentry-dart/issues/1623
There is sadly nothing I can do about it for the time being. Try downgrading any other libraries in your project which depend on uuid 4.x

ThexXTURBOXx commented:
@jhomlala Interested in me maintaining this plugin? You could add me as a collaborator.
I am planning on releasing something like `catcher_2` soon in order to keep this really important plugin properly maintained.

calcitem commented:
Dear @jhomlala and community,

I want to express my support for @ThexXTURBOXx's initiative to maintain the Catcher project. As someone who has benefited greatly from this library, it's evident that it is an invaluable resource for the Flutter community. I've seen the pull request (#234) submitted by @ThexXTURBOXx, and I believe it provides essential updates and fixes that can enhance the project significantly.

I strongly encourage you, @jhomlala, to consider adding @ThexXTURBOXx as a collaborator on this project, given their commitment to updating it. Allowing passionate and active contributors like @ThexXTURBOXx to step in would be a win for everyone involved if you cannot manage the project for any reason.

Thank you for your time and consideration.

ThexXTURBOXx committed:
Update Gradle again

ThexXTURBOXx committed:
Support older Gradle versions again

ThexXTURBOXx committed:
Update SDK versions

ThexXTURBOXx committed:
Update task registration

ThexXTURBOXx committed:
Update example dependencies

Alexqwesa committed:
Update russian localization_options.dart

btw, why pageReportModelAccept: 'Accept' instead of 'Send'? i think `send`  is better....   the same for dialogReportModelAccept

ThexXTURBOXx committed:
Rebrand to Catcher 2

ThexXTURBOXx committed:
Remove unnecessary file

ThexXTURBOXx committed:
Add files back

ThexXTURBOXx committed:
Fix other things as well

ThexXTURBOXx committed:
Fix pub badge

ThexXTURBOXx commented:
## Big news!

For anyone listening here: A few minutes ago, I have successfully released `catcher_2` version `1.0.0-alpha.0`.
It contains all the changes that have been made in this PR over the months and even a few more.
I expect some bugs to occurr since I cannot test anything on Apple-devices (I don't have any).
And I am too lazy to test _everything_, so if you encounter bugs, feel free to report them [here](https://github.com/ThexXTURBOXx/catcher_2/issues)!
Otherwise, thank you very much for the support and have a nice day!

ThexXTURBOXx committed:
Add funding

ThexXTURBOXx committed:
Add disclaimer

ThexXTURBOXx committed:
Add security policy

ThexXTURBOXx committed:
Add alpha migration guide

ThexXTURBOXx committed:
Better issue templates

ThexXTURBOXx committed:
Add missing thing to migration guide

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Better ignores

ThexXTURBOXx committed:
Merge remote-tracking branch 'solidmetanoia/master'

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Reformat CI file

ThexXTURBOXx committed:
Add CI badge to README

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Update dependencies and fix lint

ThexXTURBOXx committed:
Add additional error handlers

Should fix #11

ThexXTURBOXx committed:
Add Flutter version to issue template

ThexXTURBOXx committed:
Change initialisation order

Should fix #10

ThexXTURBOXx committed:
Add more documentation

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Improve documentation

ThexXTURBOXx committed:
Add error resilience to calls to 3rd party libraries

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Fix example and bump its dependencies

ThexXTURBOXx committed:
Even more error resilience for 3rd party libraries

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Remove unnecessary entry from issue template

narumi147 committed:
Fix null path for screenshot manager

ThexXTURBOXx committed:
Cleanup in many places

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Migrate bug reports to forms

jhomlala commented:
Thank you so much for this @ThexXTURBOXx. I have been busy for couple of months due to personal life changes. During this time you have been doing amazing job with this PR.

I don't want to close this PR, because there's a lot of important fixes. I can't ask you for more, but if you could find some time and check if there's anything that I could bring to the catcher then I would be very happy.

ThexXTURBOXx commented:
Welcome back @jhomlala! I think you have already fixed the most important issues.
I will just continue maintaining my fork and if you find that there is anything that you want to include in your `catcher` package, feel free to cherry-pick! :)

ThexXTURBOXx committed:
Update to Flutter 3.16.0

Remove all deprecations except for any PopScope-related ones. The new PopScope is way to buggy!

ThexXTURBOXx committed:
Allow new package_info_plus versions

ThexXTURBOXx committed:
Ignore WillPopScope deprecations for now

Migration is easy, but breaks backwards-compatbility... Thanks, Flutter!

ThexXTURBOXx committed:
Bump version and add changelog

m7mdra committed:
Add arabic LocalizationOptions

ThexXTURBOXx committed:
Merge upstream changes

ThexXTURBOXx committed:
Merge remote-tracking branch 'm7mdra/ar-localization_options'

ThexXTURBOXx committed:
Merge remote-tracking branch 'Alexqwesa/patch-1'

ThexXTURBOXx committed:
Bump version and add changelog

MilovdZee committed:
Fix stacktrace in console handler

ThexXTURBOXx committed:
Fix stacktrace in console handler

ThexXTURBOXx committed:
Bump version and add changelog

anilaydinn committed:
Changelog:
- Add Turkish localization options

anilaydinn committed:
Changelog:
- Add Turkish text to README.md

ThexXTURBOXx committed:
Clean up report mode support checking

Commit:
add screenshot support for Slack Handler

Commit:
add screenshot support for Slack Handler

ThexXTURBOXx committed:
Optimise screenshot handling on Slack

ThexXTURBOXx committed:
Add arabic localization options to README

ThexXTURBOXx committed:
Merge remote-tracking branch 'anilaydinn/add-turkish-localization-options'

ThexXTURBOXx committed:
Bump version and add changelog

Commit:
Adding support to send screenshot to sentry event logging

Commit:
Cleaning up screenshot file after submission to sentry

Commit:
Adding runZonedGuarded for web platform to catch asynchronous exceptions in web environment

mikeesouth committed:
Merge pull request #1 from kattalo/feature/run-zone-guarded-for-async

Adding runZonedGuarded for web platform to catch asynchronous excepti…

Commit:
Fixing timing of init widgets binding so that both sync and async exceptions are caught for both web and native platforms

mikeesouth committed:
Merge pull request #2 from kattalo/fix/init_widget_binding_timing

Fixing timing of init widgets binding to catch async errors in web

Commit:
Fixing sentry release syntax by using an @ sign between appname and version instead of a space

mikeesouth committed:
Merge pull request #3 from kattalo/fix/sentry-release-name

Fixing sentry release syntax by using an @ sign between appname and version

ThexXTURBOXx committed:
Fix directive formatting

ThexXTURBOXx committed:
Bump version and add changelog

Commit:
Adding better exception message

Commit:
Adding configurable server and logger names

mikeesouth committed:
Merge pull request #4 from kattalo/feature/improved-message-and-configurable-server-name

Feature/improved message and configurable server name

Commit:
Lowercasing appname

mikeesouth committed:
Merge pull request #5 from kattalo/fix/lowercase-appname-in-sentry-handler

Lowercasing appname

ThexXTURBOXx committed:
Support custom file names during runtime

Fixes #21

ThexXTURBOXx committed:
Allow file supplier to use info from report

Fixes #22

ThexXTURBOXx committed:
Update CMakeLists

ThexXTURBOXx committed:
Allow sentry versions 8.x

ThexXTURBOXx committed:
Bump version and add changelog

mrclauss committed:
Update version and fix some typos

ThexXTURBOXx committed:
Undo version upgrade and fix a few more things

ThexXTURBOXx committed:
Fix linter rules for Dart 3.3.0

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Update example to newest standards

ThexXTURBOXx committed:
Fix FileHandler not sending reports

ThexXTURBOXx committed:
Clean up FileHandler

ThexXTURBOXx committed:
Bump version and add changelog

bartektartanus committed:
Update package_info_plus to v6

ThexXTURBOXx committed:
Allow device_info_plus 10.0.0 to be used

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Update dependencies

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Allow package_info_plus versions 8.x

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Remove direct dependency on platform interface

ThexXTURBOXx committed:
Migrate Slack screenshots to new API calls

ThexXTURBOXx committed:
Remove superfluous log

ThexXTURBOXx committed:
Better example structure

ThexXTURBOXx committed:
Update flutter_lints

ThexXTURBOXx committed:
Migrate screenshots to XFiles

ThexXTURBOXx committed:
Workaround for error catching on Web

mikeesouth committed:
Merge from upstream 20240520 (#6)

* Fix directive formatting

* Bump version and add changelog

* Support custom file names during runtime

Fixes #21

* Allow file supplier to use info from report

Fixes #22

* Update CMakeLists

* Allow sentry versions 8.x

* Bump version and add changelog

* Update version and fix some typos

* Undo version upgrade and fix a few more things

* Fix linter rules for Dart 3.3.0

* Bump version and add changelog

* Update example to newest standards

* Fix FileHandler not sending reports

* Clean up FileHandler

* Bump version and add changelog

* Update package_info_plus to v6

* Allow device_info_plus 10.0.0 to be used

* Bump version and add changelog

* Update dependencies

* Bump version and add changelog

* Allow package_info_plus versions 8.x

* Bump version and add changelog

* Remove direct dependency on platform interface

* Migrate Slack screenshots to new API calls

* Remove superfluous log

* Better example structure

* Update flutter_lints

* Migrate screenshots to XFiles

* Workaround for error catching on Web

---------

Co-authored-by: Nico Mexis <nico.mexis@kabelmail.de>
Co-authored-by: Martin Clauss <m@clauss.email>
Co-authored-by: Bartlomiej Tartanus <bartektartanus@gmail.com>
Co-authored-by: Michael Söderman <micke@kattalo.com>

Commit:
Merge remote-tracking branch 'upstream/master' into merge-from-upstream-20240520-2

mikeesouth committed:
Merge pull request #7 from kattalo/merge-from-upstream-20240520-2

Merge from upstream 20240520 2

ThexXTURBOXx committed:
Merge remote-tracking branch 'kattalo/master'

ThexXTURBOXx committed:
Merge #32 and improve a few things

ThexXTURBOXx committed:
Merge remote-tracking branch 'jhomlala/master'

ThexXTURBOXx committed:
Prepare for 2.0.0 alpha

ThexXTURBOXx committed:
Merge remote-tracking branch 'jhomlala/master'

Again...

ThexXTURBOXx committed:
Apply proper theme colours

Fixes #31

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Fix analyser issues

ThexXTURBOXx committed:
Update fluttertoast to 8.2.6

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Better logging

Fixes #37

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Fix example on Android

JDK 21 was causing issues...

ThexXTURBOXx committed:
Update dependencies

ThexXTURBOXx committed:
Fix lint

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Fix logger after updateConfig

Fixes #39

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Add extra data parameter to reportCheckedError

Fixes #40

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Update flutter_mailer to 3.x

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Fix lint

comment_references is broken currently...

ThexXTURBOXx committed:
Update example dependencies

ThexXTURBOXx committed:
Allow sentry versions 9.x

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Recommend standard 2.0.0 release, not alpha anymore

ThexXTURBOXx committed:
Fix CI by migrating away from sketchy container

IVLIVS-III committed:
Support `go_router` / `MaterialApp.router(…)` (#48)

* Exposed `configureNavigatorKey` to update the used `navigatorKey` after `runApp` was called.

This allows combining `go_router` / `MaterialApp.router` with the `DialogReportMode`.

* Refactored to remove abstractions.

ThexXTURBOXx committed:
Bump version and add changelog

mrclauss committed:
ReportLog: Relax restriction in device parameter reporting (#49)

ThexXTURBOXx committed:
Update linter

ThexXTURBOXx committed:
Update CI

ThexXTURBOXx committed:
Allow newer dependency versions

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Update example dependencies

ThexXTURBOXx committed:
Allow fluttertoast 9.x versions

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Update dependencies

ThexXTURBOXx committed:
Cleanup code a bit

ThexXTURBOXx committed:
Update checkout to v6

ThexXTURBOXx committed:
Still also allow mailer 6.x

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Do not print empty sections in ConsoleHandler

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Fix critical bug introduced in 2.1.7

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Add JNI plugin to example

ThexXTURBOXx committed:
Allow newer dependency versions

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Allow fluttertoast 10.x versions

ThexXTURBOXx committed:
Update CI

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Allow sentry 10.x versions

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Fix lint

ThexXTURBOXx committed:
Bump version and add changelog

ThexXTURBOXx committed:
Update example libraries
